keyboards/ergodash/rev1/keymaps/m47ch4n/keymap.c

```diff
@@ -0,0 +1,226 @@
+     * |      | Home |PageDn|PageUp|  End ||||||||      |                    |      | Left | Down |  Up  | Right||||||||      |
+     * |------+------+------+------+------+------+---------------------------+------+------+------+------+------+------+------|
+     * |      |      |      |      |      |      |      |                    |      |      |      |   {  |   }  |      |      |
+     * |-------------+------+------+------+------+------+------+------+------+------+------+------+------+------+-------------|
+     * |      |      |      |      ||||||||      |      |      ||||||||      |      |      ||||||||      |      |      |      |
+     * ,----------------------------------------------------------------------------------------------------------------------.
+     */
+    [_ADJUST] = LAYOUT(
+      _______, _______, _______, _______, _______, _______, US     ,                        JP,      _______, _______, _______, _______, _______, _______, \
+      _______, _______, _______, _______, _______, _______, _______,                        _______, _______, _______, _______, _______, _______, _______, \
+      _______, KC_HOME, KC_PGDN, KC_PGUP, KC_END,  XXXXXXX, _______,                        _______, KC_LEFT, KC_DOWN, KC_UP,   KC_RGHT, XXXXXXX, _______, \
+      _______, _______, _______, _______, _______, _______, _______,                        _______, _______, _______, LCBR,    RCBR,    _______, _______, \
+      _______, _______, _______, _______,          _______, _______,_______,        _______,_______, _______,          _______, _______, _______, _______  \
+    )};
+// clang-format on
+
+#ifdef AUDIO_ENABLE
+float tone_qwerty[][2] = SONG(QWERTY_SOUND);
+#endif
+
+#define JP_LAYOUT true
+#define US_LAYOUT false
+
+#ifdef MASTER_LEFT
+bool LAYOUT_STATUS = US_LAYOUT;
+#else
+bool LAYOUT_STATUS = JP_LAYOUT;
+#endif
+
+bool SHIFT_PRESSED = false;
+
+void persistent_default_layer_set(uint16_t default_layer) {
+    eeconfig_update_default_layer(default_layer);
+    default_layer_set(default_layer);
+}
+
+#define SEND_STRING_RESTORE(STR) (SHIFT_PRESSED ? SEND_STRING(STR SS_DOWN(X_LSHIFT)) : SEND_STRING(STR SS_UP(X_LSHIFT)))
+
+#define KEY(CODE) (record->event.pressed ? SEND_STRING(SS_DOWN(X_##CODE)) : SEND_STRING_RESTORE(SS_UP(X_##CODE)))
+
+#define KEY_SHIFT(CODE) (record->event.pressed ? SEND_STRING(SS_DOWN(X_LSHIFT) SS_DOWN(X_##CODE)) : SEND_STRING_RESTORE(SS_UP(X_##CODE)))
+
+#define KEY_UPSHIFT(CODE) (record->event.pressed ? SEND_STRING(SS_UP(X_LSHIFT) SS_DOWN(X_##CODE)) : SEND_STRING_RESTORE(SS_UP(X_##CODE)))
+
+#define SHIFT_DU(CODE_DOWN, CODE_UP) (SHIFT_PRESSED ? CODE_DOWN : CODE_UP)
+#define CASE_US(CODE, US, JP)                   \
+    case CODE:                                  \
+        (LAYOUT_STATUS == JP_LAYOUT ? JP : US); \
+        return false;
+
+bool pushed_lower_l = false;
+bool pushed_lower_r = false;
+bool pushed_adjust  = false;
+
+void update_layer(void) {
```

Comment:
I think you may want to do this, instead: 
https://docs.qmk.fm/#/ref_functions?id=update_tri_layer_statestate-x-y-z

I think you're trying to do the same thing, and this is a much simpler way to do so.

Reply:
I fixed my code using `update_tri_layer` instead.
You can see my changes here https://github.com/qmk/qmk_firmware/pull/6673/commits/ba9fc0ee6c5b34f922e018f8860296ac288d467a.